src/aws-cpp-sdk-core/source/http/curl/CurlHttpClient.cpp

```diff
@@ -272,6 +275,14 @@ static size_t WriteHeader(char* ptr, size_t size, size_t nmemb, void* userdata)
         {
             response->AddHeader(StringUtils::Trim(keyValuePair[0].c_str()), StringUtils::Trim(keyValuePair[1].c_str()));
         }
+        //checking for end of all the headers before setting response code
+        else if (headerLine == "\r\n" && context->m_curlHandle)
+        {
+            long responseCode{-1};
+            curl_easy_getinfo(context->m_curlHandle, CURLINFO_RESPONSE_CODE, &responseCode);
+            response->SetResponseCode(static_cast<HttpResponseCode>(responseCode));
```

Comment:
feels kinda weird to set [response code twice](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/source/http/curl/CurlHttpClient.cpp#L852) i would prefer if we set it once here, and then assert that its set before leaving the function